### ISSUE TITLE
Certbot renewal does not use the last settings

### ISSUE THREAD

**Issue by jdecalog** [certbot/certbot#4887]
## My operating system is (include version):
Debian 8.6

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
Certbot is installed via Debian backport and is up to date (certbot 0.10.2)

## I ran this command and it produced this output:
certbot renew failed because the port 443 is already in use (haproxy)
The trace is :
```
Processing /etc/letsencrypt/renewal/mycert.conf
-------------------------------------------------------------------------------
Cert is due for renewal, auto-renewing...
Starting new HTTPS connection (1): acme-v01.api.letsencrypt.org
Renewing an existing certificate
Performing the following challenges:
tls-sni-01 challenge for mycert
-------------------------------------------------------------------------------
The program haproxy (process ID 20526) is already listening on TCP port 443.
This will prevent us from binding to that port. Please stop the haproxy program
temporarily and then try again. For automated renewal, you may want to use a
script that stops and starts your webserver. You can find an example at
https://certbot.eff.org/docs/using.html#renewal . Alternatively you can use the
webroot plugin to renew without needing to stop and start your webserver.
-------------------------------------------------------------------------------
Cleaning up challenges

```
## Certbot's behavior differed from what I expected because:
The settings in the renewal config file are :
```
# Options used in the renewal process
[renewalparams]
authenticator = standalone
installer = None
account = xxxxxxxxxx
http01_port = 54321
```

If is run the following :
```
certbot certonly --text -d mycert --standalone --preferred-challenges http-01 --http-01-port 54321
```
it works fine (cert has been generated with this command)

**Comment by jdecalog** [comment]
The command worked when added in the renewall conf file the following :

`pref_challs = http-01`
why the line is not added a the creation (certbot was in a previous version when creating the certs)

**Comment by SwartzCr** [comment]
That's really interesting - glad to know that you have it working now.
Do you know if you ever successfully completed a renewal? Did you update the package recently?
I'm curious if @bmw has any insight.

**Comment by bmw** [comment]
When `--preferred-challenges` was added in Certbot 0.9.0, there was a bug where the value was not saved for renewal. This was [fixed in Certbot 0.10.2](https://github.com/certbot/certbot/blob/master/CHANGELOG.md#0102---2017-01-25). If you last issued the certificate with one of the older versions of the client, this could happen.

If you run the same `certbot certonly ...` command you provided above with Certbot 0.10.2+ to obtain a certificate, the `--preferred-challenges` value should be saved. Keep in mind the renewal configuration file is only updated if you actually obtain a certificate. If you select option 1 at the prompt below, the file will not be modified:
```
You have an existing certificate that has exactly the same domains or certificate name you requested and isn't close to expiry.
(ref: /etc/letsencrypt/renewal/mycert.conf)

What would you like to do?
-------------------------------------------------------------------------------
1: Keep the existing certificate for now
2: Renew & replace the cert (limit ~5 per 7 days)
-------------------------------------------------------------------------------
Select the appropriate number [1-2] then [enter] (press 'c' to cancel):
```
Sorry for the trouble!

**Comment by jdecalog** [comment]
Ok I understand now

I think the version at the first time was 0.9.4

I have fixed my puppet module for the deployement, by forcing the line if missing.
For now the next installation would be correct as the version will be upper than 0.10.2
